### DISCUSSION TITLE
Course Timeline and Fee

### DISCUSSION THREAD

**Discussion by shubhankur** [snigi-gupta/AttendingUniversityAtBuffalo#67] [General]
Hi Snigdha,

I am really thankful for all the information. But, I do have a few doubts and would be grateful if you could address them.

1. What is the timeline to complete our degree? I believe we can complete our degree in 1.5 years only, so the 3-month deadline to search for the job starts then only?

2. Can we do a full-time internship/co-op but remotely?

3. What is the exact tuition fee for the entire program? 

4. What is the minimum number of credits that we have to do each semester?

Thank You

**Comment by snigi-gupta** [comment]
@shubhankur 
1-. What is the timeline to complete our degree? I believe we can complete our degree in 1.5 years only, so the 3-month deadline to search for the job starts then only?
Ans: For an MS degree without a thesis you have to graduate in 1.5 yrs. Unless you do not meet the graduate requirements (for example you have not fulfilled all the required credits in the required categories) or you have a cumulative GPA < 3.0 you cannot extend your MS duration at UB). For an MS degree with a thesis, you can graduate in 2 yrs provided your thesis is accepted by the prof and the board.
For the second part of your ques, let me explain by example. 
Suppose you join UB during Fall 2022 (i.e Aug 2022), then you will officially graduate in Spring 2024 (Feb 1). You can start looking for jobs 1 yr to 6 months in advance (because that is how it works in the US, usually students in this batch start looking for jobs in Aug 2023). After your graduation, you would have 90 days to look for a job from the starting EAD date that you mentioned. So let's say you chose your EAD date as March 15th (you can choose an EAD date b/w Feb 1 and Apr 1 [60 days from your graduation date]), then you would have 90 days to look for a job from March 15th. All this might feel complex, but I would suggest you not plan so far in the future. The 3-month deadline, in this case, is basically  March 15th - June 15th. 

2- Can we do a full-time internship/co-op but remotely?
Ans: Yes, you can. Depends on the company. I did an internship and a co-op both remotely. 

3- What is the exact tuition fee for the entire program? 
Ans: I think you can find an accurate answer than mine by doing a simple google search: [Financial Form](https://www.buffalo.edu/content/dam/www/internationaladmissions/documents/Financial_Form_Standard_Graduate-2022.pdf), [Link](https://www.buffalo.edu/grad/explore/funding/cost.html)
I think during my time it was $42,452 (out of this tuition was $27,387)

4- What is the minimum number of credits that we have to do each semester?
Ans: 12 credits per sem. Last sem only 6 credits. Total 30 creds. Read more about that [here](https://github.com/snigi-gupta/AttendingUniversityAtBuffalo/tree/main/Computer%20Science%20Important%20Tips#how-to-complete-ms-requirements)

**Reply by shubhankur** [reply]
Hi Snigdha,

I really appreciate your detail reply. Sorry to bother you but I have a few subsequent doubts.

1. I understand when you say that not to plan so far, but I have a little anxiety regarding that issue. As far as I know, in general scenario we have 2 years + 3 years period to stay in US. And our company applies for our H1B 3 times, once per year during the time of Jan-March each year. But I am still a little confused for the 1.5years case. Our total stay period will be 1.5+3 or 1.5+3.5 ?

2. Also, can you explain a little more about EAD ?

3. Can we choose credits like 4+4+1+1 if we want to extend to the 4th semester or is there any other way without thesis to extend it to 4 semesters?

4. When we do summer internship and suppose our company wants us to continue, will buffalo allow us to continue it in the 3rd semester ? If yes then what are the terms and conditions and for how long can we extend that internship?

Thank You

**Reply by snigi-gupta** [reply]
@shubhankur 
1- 
I understand, that things can get complicated in understanding these things, but let me try to simplify it for you.
The purpose of you doing the master's is for you to get a job. So whether you do 1.5 yrs of MS or 2 yrs of MS you will have additional 3 yrs to work on your F1 visa (your student visa). So nothing changes because of the duration of the course.
I graduated in Spring 2021 (Feb1) and I get 3 attempts for the H1 lottery. Whether you do a 2 yr or 1.5 yr course, it does not matter.
If you do 2 yrs course your total stay is 2 yrs + 3 yrs. If you do 1.5 yrs course, your total stay is 1.5yrs + 3yrs. 
I chose to do a 1.5yrs course because I had a loan, I wanted to graduate sooner, get a job sooner, and pay back my loans sooner. Extending my master's would not have given me any benefit. 
I am curious, what is your concern here? Are you afraid to miss out on something if you choose a 1.5 yrs of course? Or you are concerned you won't get 3 yrs after MS to work in the USA? Because irrespective of what you choose, if you are doing a STEM course, you will get 3 yrs to work on F1 visa. This part is not related to the duration of your MS.

2-
EAD is Employment Authorization Document. It allows you to work after your graduation using this card. You need to apply for this card before you graduate, so when you start your new job, you provide this document as proof that you are legally allowed to work in the USA.
You mention a date on this document, by which you are saying, "hey, I graduate on Feb 1, I think I will be able to get a job by x date (say Mar15, if I am unable to get a job by then, then I have 90 days to try harder, if I cannot get a job in your country, I will exit your country". 

3- 
No, you can have a reduced course load only once. Without a thesis, you can do this, if you deliberately pick courses that do not satisfy your graduation checklist. In that case, the university will say "hey you have not picked 1 course from each category, or you still need to do 1 more course in your specialized area, pay us another sem, to get your graduation degree". But I don't see why you will pay more to the univ deliberately to extend your graduation. What benefit do you think you will have?

4-
I think I already answered the co-op thing. I did a summer internship, and I continued my internship during my 3rd sem. You just have to make sure your internship/co-op period does not extend for more than 12 months, otherwise, it gets deducted from your OPT months (the 3 yrs you have, which you don't want because you want to use all the 3 yrs in doing a job and not an internship). Also, there are department-related rules too like the internship should be related to your field, or the no. of credits you get, but for that, you should ask the department.

Again, no point in trying to understand all these technical terms from sitting in India. Decide your university based on the curriculum, the finance, and your interest. That should be your criteria for deciding on a university. 
Everything else then starts happening automatically. Each univ have their own term and conditions and they are subject to change. So decide your university first and you will have plenty of time to understand the jargons, once you are in the USA.
